packages/block-editor/src/hooks/content-lock-ui.js

```diff
@@ -37,7 +33,7 @@ function StopEditingAsBlocksOnOutsideSelect( {
 		if ( ! isBlockOrDescendantSelected ) {
 			stopEditingAsBlock();
 		}
-	}, [ isBlockOrDescendantSelected ] );
+	}, [ isBlockOrDescendantSelected, stopEditingAsBlock ] );
```

Comment:
Fixes the ESLin warning. The `stopEditingAsBlock` callback is memoized.

Reply:
The `stopEditingAsBlock` callback has a lot of dependencies, but they are mostly `useDispatch` return values which should never change. Then there's `clientId`, also constant, and a constant `focusModeToRevert` ref. I wonder if the ref really needs to be a dependency, in practice it has no effect and I think the ESLint rule will forgive an undeclared dependency that's a return value of `useRef` and is accessed as `.current`.

Reply:
Unfortunately, we have to declare callbacks returned by `useDispatch` to satisfy the linter, but as you said, they should have a stable reference.

I'll remove `focusModeToRevert` from dependencies.